### PR TITLE
Used HashSet in bfs.rs

### PR DESCRIPTION
**Pull request by cherrygot-personal**
Optimization required in issue https://github.com/purpleprotocol/graphlib/issues/6

### CODE REVIEW

**Comment by octavonce** [comment]
Nice, this uses less code than my proposed solution. Thank you for opening the PR!

File: src/iterators/bfs.rs [hunk 1]
```diff
@@ -28,7 +29,7 @@ impl<'a, T> Bfs<'a, T> {
         Bfs {
             queue: VecDeque::with_capacity(graph.vertex_count()),
             current_ptr,
-            visited_stack: Vec::with_capacity(graph.vertex_count()),
+            visited_set: HashSet::new(),
```
Comment (octavonce) [inline src/iterators/bfs.rs]: Maybe we can say here `HashSet::with_capacity(graph.vertex_count)` to avoid reallocation?

**Review by octavonce — CHANGES_REQUESTED** [review]
Everything else looks great!

File: src/iterators/bfs.rs (hunk 1 above)
Comment (cherrygot-personal) [reply src/iterators/bfs.rs]: Yeah. That's much better.

**Review by octavonce — APPROVED** [review]

**Pull request merged**